service/controller/v1/resource/deployment/resource.go

```diff
@@ -90,6 +84,13 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 		return nil, microerror.Mask(err)
 	}
 
+	// Deleting the resource group will take care about cleaning
+	// deployments.
```

Comment:
A bit more information on why we cancel would be cool. IIRC there was a nice comment already in your previous commit containing these information.